lib/src/ui/pages/profile/profile_page.dart

```diff
@@ -1,12 +1,61 @@
 import 'package:allthenews/generated/l10n.dart';
+import 'package:allthenews/src/di/injector.dart';
+import 'package:allthenews/src/ui/pages/profile/auth/authentication_page.dart';
 import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+import 'auth/authorization_notifier.dart';
+
+abstract class _Constants {
+  static const fontFamily = 'Chomsky';
+}
+
+class ProfilePage extends StatefulWidget {
+  @override
+  _ProfilePageState createState() => _ProfilePageState();
+}
+
+class _ProfilePageState extends State<ProfilePage> {
+  final AuthorizationNotifier _authorizationNotifier = inject<AuthorizationNotifier>();
+
+  @override
+  void initState() {
+    super.initState();
+    _authorizationNotifier.getCurrentUser();
+  }
 
-class ProfilePage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
-    return Container(
-      alignment: Alignment.center,
-      child: Text(Strings.of(context).profile),
-    );
+    return ChangeNotifierProvider.value(
+        value: _authorizationNotifier,
+        builder: (providerContext, child) {
+          final viewState =
+              providerContext.select((AuthorizationNotifier notifier) => notifier.state);
+
+          if (viewState.user == null) {
```

Comment:
można by było zrobić jakoś bardziej generycznie bo jak domniemam, każdy ekran, który będziemy mieli w profilu będzie musiał sprawdzać czy użytkownik jest zalogowany? Więc może jakiś abstrakcyjny notifier i page dla tych ekranów? :)

Reply:
co masz na myśli pisząc "każdy ekran w profilu"? zakładałem, że to będzie tylko jeden widok. Tak czy inaczej rzeczy związane z profilem zostawiłbym na taska z profilem

Reply:
chodzi mi o to, że z poziomu profilu będziemy przechodzić na inne zakładki i one też będą autoryzowane, bo w teorii może się zdarzyć, że ktoś zostanie na tym ekranie, apkę zrzuci na stos, potem wróci i już nie będzie zalogowany? Może być osobno przy pierwszym ekranie profilowym, jak najbardziej :)

Reply:
może się tak zdarzyć jakby usunąć usera ręcznie z firebase i chyba jakby się ktoś w międzyczasie na innym device zalogował, ale nie sprawdzałem tego w sumie